javalib/src/main/scala/java/lang/System.scala

```diff
@@ -39,6 +41,38 @@ object System {
 
   def gc(): Unit = GC.collect()
 
+  private lazy val envVars: Map[String, String] = {
+    // workaround since `while(ptr(0) != null)` causes segfault
+    def isDefined(ptr: Ptr[CString]): Boolean = {
```

Comment:
Made a comment on #367 since it looks like the reason this workaround is needed.